dotcom-rendering/src/web/components/Links.importable.tsx

```diff
@@ -135,34 +134,37 @@ const linksStyles = css`
 	}
 `;
 
-export const Links = ({
-	userId,
-	supporterCTA,
-	idUrl: idUrlFromConfig,
-	mmaUrl: mmaUrlFromConfig,
-}: Props) => {
-	const [showSupporterCTA, setShowSupporterCTA] = useState<boolean>();
-	const [userIsDefined, setUserIsDefined] = useState<boolean>();
-
-	// show supporter CTA if support messaging isn't shown
-	useEffect(() => {
-		setShowSupporterCTA(
-			getCookie({
-				name: 'gu_hide_support_messaging',
-				shouldMemoize: true,
-			}) === 'true',
-		);
-	}, []);
-
-	// we intentionally re-render here because we know the DOM structure could be different
-	// from the server rendered version. This forces a full validation
-	useOnce(() => {
-		setUserIsDefined(!!userId);
-	}, [userId]);
+const SignIn = ({ idUrl }: { idUrl: string }) => (
+	<a
+		css={linkStyles}
+		href={`${idUrl}/signin?INTCMP=DOTCOM_NEWHEADER_SIGNIN&ABCMP=ab-sign-in&${createAuthenticationEventParams(
+			'guardian_signin_header',
+		)}`}
+		data-link-name="nav2 : topbar : signin"
+	>
+		<ProfileIcon /> Sign in
+	</a>
+);
 
-	// Fall back on prod URLs just in case these aren't set for any reason
-	const idUrl = idUrlFromConfig || 'https://profile.theguardian.com';
-	const mmaUrl = mmaUrlFromConfig || 'https://manage.theguardian.com';
+const MyAccount = ({
+	mmaUrl,
+	idUrl,
+	discussionApiUrl,
+}: {
+	mmaUrl: string;
+	idUrl: string;
+	discussionApiUrl: string;
+}) => {
+	const { data } = useApi<{ userProfile: UserProfile }>(
+		joinUrl(discussionApiUrl, 'profile/me?strict_sanctions_check=false'),
+		{},
+		{
+			credentials: 'include',
+		},
+	);
+
+	// If we don't have user data display sign in to the user. SWR will retry in the background if the request failed
+	if (!data?.userProfile?.userId) return <SignIn idUrl={idUrl} />;
```

Comment:
+1 for the comment

Should we also check for `error` here? This is returned from `useApi`

Reply:
We _could_ use a loading state here. It's not something for this PR but it might be more truthful that showing `SignIn` while we wait for the api call.

Just something to consider, no action needed

Reply:
> Should we also check for error here? This is returned from useApi

We could do, it would be relatively safe to assume that it wouldn't change anything (in the sense that if there's an error, data.userProfile.userId wouldn't be set), but I think it is good practice to check anyway so I will add it!